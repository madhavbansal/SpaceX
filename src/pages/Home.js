import React, { useEffect, useState } from 'react';
import { Filters, LaunchPrograms } from './index';
import { useDevice } from '../customHooks'
import { constants } from '../utils/constant';
import * as styles from './Home.style';

const Home = () => {
    const deviceType = useDevice();
    const style = styles[deviceType];
    return (
        <>
            <div style={style.text}>{constants.spaceXPrograms}</div>
            <div style={style.common}>
                <div style={style.filters}>
                    <Filters/>
                </div>
                <div style={style.LaunchPrograms}>
                    <LaunchPrograms/>
                </div>
            </div>
        </>
    )


}

export default Home