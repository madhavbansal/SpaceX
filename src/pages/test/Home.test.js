import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';

jest.mock('../Filters', ()=> 'Filters')
jest.mock('../LaunchPrograms', ()=> 'LaunchPrograms')

test('Test the Home Component', () => {
  const component = renderer.create(
    <Home/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});