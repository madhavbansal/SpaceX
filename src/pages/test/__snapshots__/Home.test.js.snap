// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home Component 1`] = `
Array [
  "SpaceX Launch Programs",
  <Filters />,
  <LaunchPrograms />,
]
`;
